log/keyvalue.go

```diff
@@ -385,3 +386,58 @@ func Empty(key string) KeyValue {
 func (a KeyValue) String() string {
 	return fmt.Sprintf("%s:%s", a.Key, a.Value)
 }
+
+// ConvertAttributeValue converts [attribute.Value) to [Value].
+func ConvertAttributeValue(value attribute.Value) Value {
+	switch value.Type() {
+	case attribute.INVALID:
+		return Value{}
+	case attribute.BOOL:
+		return BoolValue(value.AsBool())
+	case attribute.BOOLSLICE:
+		val := value.AsBoolSlice()
+		res := make([]Value, 0, len(val))
+		for _, v := range val {
+			res = append(res, BoolValue(v))
+		}
+		return SliceValue(res...)
+	case attribute.INT64:
+		return Int64Value(value.AsInt64())
+	case attribute.INT64SLICE:
+		val := value.AsInt64Slice()
+		res := make([]Value, 0, len(val))
+		for _, v := range val {
+			res = append(res, Int64Value(v))
+		}
+		return SliceValue(res...)
+	case attribute.FLOAT64:
+		return Float64Value(value.AsFloat64())
+	case attribute.FLOAT64SLICE:
+		val := value.AsFloat64Slice()
+		res := make([]Value, 0, len(val))
+		for _, v := range val {
+			res = append(res, Float64Value(v))
+		}
+		return SliceValue(res...)
+	case attribute.STRING:
+		return StringValue(value.AsString())
+	case attribute.STRINGSLICE:
+		val := value.AsStringSlice()
+		res := make([]Value, 0, len(val))
+		for _, v := range val {
+			res = append(res, StringValue(v))
+		}
+		return SliceValue(res...)
+	}
+	// This code should never be reached
+	// as log attributes are a superset of standard attributes.
+	panic("unknown attribute type")
+}
+
+// ConvertAttributeKeyValue converts [attribute.KeyValue) to [KeyValue].
+func ConvertAttributeKeyValue(kv attribute.KeyValue) KeyValue {
```

Comment:
Are the call sites for this expected to only have one attribute? Should there be a `func([]attribute.KeyValue) []KeyValue` conversion function?

Reply:
Hard to tell. I decided to postpone this as creating a slice conversion function is very easy with the provided function.